batch/batch/worker/worker.py

```diff
@@ -3031,9 +3035,11 @@ def return_jvm(self, jvm: JVM):
     async def return_broken_jvm(self, jvm: JVM):
         return await self._jvmpools_by_cores[jvm.n_cores].return_broken_jvm(jvm)
 
-    @property
-    def headers(self) -> Dict[str, str]:
-        return {'X-Hail-Instance-Name': NAME, 'X-Hail-Instance-Token': self.instance_token}
+    async def headers(self):
+        headers = {'X-Hail-Instance-Name': NAME, 'X-Hail-Instance-Token': self.instance_token}
+        if isinstance(CLOUD_WORKER_API, TerraAzureWorkerAPI):
+            headers['Authorization'] = 'Bearer: ' + await CLOUD_WORKER_API.vm_identity_token()
```

Comment:
So, this is for talking back to the driver right? And this is because the comms from worker to driver has to go through some external load balancer rather than within the local VPC's network?

Reply:
Regardless, this should probably be a `worker_api.extra_headers()` method rather than an if-isinstance.

Reply:
First comment is correct. Regarding the second comment, I'll do the `extra_headers` method.